babyagi.py

```diff
@@ -3,16 +3,22 @@
 import time
 from collections import deque
 from typing import Dict, List
+from dotenv import load_dotenv
+import os
 
-#Set API Keys
-OPENAI_API_KEY = ""
-PINECONE_API_KEY = ""
-PINECONE_ENVIRONMENT = "us-east1-gcp" #Pinecone Environment (eg. "us-east1-gcp")
+load_dotenv()
 
-#Set Variables
-YOUR_TABLE_NAME = "test-table"
-OBJECTIVE = "Solve world hunger."
-YOUR_FIRST_TASK = "Develop a task list."
+# Set API Keys
+OPENAI_API_KEY = os.getenv("OPENAI_API_KEY")
+PINECONE_API_KEY = os.getenv("PINECONE_API_KEY")
+PINECONE_ENVIRONMENT = os.getenv("PINECONE_ENVIRONMENT")
+
+# Table config
+YOUR_TABLE_NAME = os.getenv("TABLE_NAME")
+
+# Project config
+OBJECTIVE = os.getenv("OBJECTIVE")
+YOUR_FIRST_TASK = os.getenv("FIRST_TASK")
```

Comment:
could use defaults here.  would also be nice to throw in some `assert`s so that if your API key is blank, we early exit.

Reply:
`# Set API Keys
OPENAI_API_KEY = os.getenv("OPENAI_API_KEY", "")
PINECONE_API_KEY = os.getenv("PINECONE_API_KEY", "")
PINECONE_ENVIRONMENT = os.getenv("PINECONE_ENVIRONMENT", "us-east1-gcp")

# Check for missing API keys
assert OPENAI_API_KEY, "OPENAI_API_KEY is missing"
assert PINECONE_API_KEY, "PINECONE_API_KEY is missing"
`

Reply:
why set those env vars to an empty string? could more explicitly `assert var is not None` if you just leave the default empty, rather than providing `""`

Reply:
The explicit assertion outweighs the extra words, so seems better.

We could also add a try-except block could help the code handle this error gracefully. (see error handling #13 )

```
import os

# Set API Keys
OPENAI_API_KEY = os.getenv("OPENAI_API_KEY")
PINECONE_API_KEY = os.getenv("PINECONE_API_KEY")
PINECONE_ENVIRONMENT = os.getenv("PINECONE_ENVIRONMENT", "us-east1-gcp")

# Check for missing API keys
try:
    assert OPENAI_API_KEY is not None, "OPENAI_API_KEY is missing"
    assert PINECONE_API_KEY is not None, "PINECONE_API_KEY is missing"
except AssertionError as e:
    # Handle the AssertionError here
    print(f"Error: {e}")
    # You could also log the error, raise a custom exception, or display a user-friendly error message, depending on your use case
    exit(1)

# If all assertions pass, continue with the rest of the code
```

Reply:
@DrewMcArthur @honeykjoule Thanks, just committed. Tell me what you think.

Since this is really just for folks to mess around with and tweak, I think optimizing for simplicity/understandability may be the best route right now. Just my two cents

Reply:
100% agree that simplicity and understandability is priority